File: app/src/main/java/com/sorien/ppbthome/MainActivity.java
package com.sorien.ppbthome;

import android.Manifest;
import android.bluetooth.BluetoothAdapter;
import android.bluetooth.BluetoothManager;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.widget.Toast;

import java.util.ArrayList;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.SwitchCompat;
import androidx.core.app.ActivityCompat;

public class MainActivity extends AppCompatActivity {

    private static final int PERMISSION_REQUEST_CODE = 1;
    private static final int BLUETOOTH_REQUEST_ENABLE = 2000;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        SwitchCompat onOffSwitch = findViewById(R.id.activate_switch_id);

        onOffSwitch.setChecked(AdvertiserService.isServiceCreated());

        onOffSwitch.setOnCheckedChangeListener((buttonView, isChecked) -> {
            if (isChecked) {
                startServiceEnableBluetooth();
            } else {
                stopService();
            }
        });
    }

    public void startServiceEnableBluetooth() {
        BluetoothManager bluetoothManager = (BluetoothManager)MainActivity.this.getSystemService(MainActivity.BLUETOOTH_SERVICE);
        if (!bluetoothManager.getAdapter().isEnabled()) {
            Intent enableBtIntent = new Intent(BluetoothAdapter.ACTION_REQUEST_ENABLE);
            ActivityCompat.startActivityForResult(MainActivity.this, enableBtIntent, BLUETOOTH_REQUEST_ENABLE, null);
        } else {
            startServiceRequestPermissions();
        }
    }

    public void startServiceRequestPermissions() {

        String[] permissions = new String[]{Manifest.permission.BLUETOOTH_ADVERTISE, Manifest.permission.BLUETOOTH_CONNECT};
        ArrayList<String> requirePermissions = new ArrayList<>();

        for (String permission : permissions) {
            if (ActivityCompat.checkSelfPermission(MainActivity.this, permission) != PackageManager.PERMISSION_GRANTED) {
                requirePermissions.add(permission);
            }
        }

        if (!requirePermissions.isEmpty()) {
            ActivityCompat.requestPermissions(MainActivity.this, requirePermissions.toArray(new String[0]), PERMISSION_REQUEST_CODE);
        } else {
            startService();
        }
    }

    public void startService() {
        startService(new Intent(MainActivity.this, AdvertiserService.class));

        Config config = new Config(this);
        config.setAutoStart(true);

        Toast.makeText(this, "Broadcasting started.", Toast.LENGTH_SHORT).show();
    }

    public void stopService() {
        if (AdvertiserService.isServiceCreated()) {
            stopService(new Intent(MainActivity.this, AdvertiserService.class));

            Config config = new Config(this);
            config.setAutoStart(false);

            Toast.makeText(this, "Broadcasting finished.", Toast.LENGTH_SHORT).show();
        }
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {

        super.onRequestPermissionsResult(requestCode, permissions, grantResults);

        if (requestCode == PERMISSION_REQUEST_CODE) {
            if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                startService();
            } else {
                SwitchCompat onOffSwitch = findViewById(R.id.activate_switch_id);
                onOffSwitch.setChecked(AdvertiserService.isServiceCreated());
            }
        }
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == BLUETOOTH_REQUEST_ENABLE){
            if (resultCode == RESULT_OK){
                startServiceRequestPermissions();
            }
            else if (resultCode==RESULT_CANCELED){
                SwitchCompat onOffSwitch = findViewById(R.id.activate_switch_id);
                onOffSwitch.setChecked(false);
                Toast.makeText(this, "Bluetooth not enabled.", Toast.LENGTH_SHORT).show();
            }
        }
    }

}

File: app/src/main/java/com/sorien/ppbthome/MainActivity.kt
package com.sorien.ppbthome

import android.Manifest
import com.sorien.ppbthome.Config.autoStart
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.sorien.ppbthome.R
import androidx.appcompat.widget.SwitchCompat
import com.sorien.ppbthome.AdvertiserService
import android.widget.CompoundButton
import androidx.core.app.ActivityCompat
import android.content.pm.PackageManager
import com.sorien.ppbthome.MainActivity
import android.content.Intent
import android.widget.Toast
import java.util.ArrayList

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val onOffSwitch = findViewById<SwitchCompat>(R.id.activate_switch_id)
        onOffSwitch.isChecked = AdvertiserService.isServiceCreated()
        onOffSwitch.setOnCheckedChangeListener { buttonView: CompoundButton?, isChecked: Boolean ->
            if (isChecked) {
                startServiceRequestPermissions()
            } else {
                stopService()
            }
        }
    }

    fun startServiceRequestPermissions() {
        val permissions = arrayOf(Manifest.permission.BLUETOOTH_ADVERTISE, Manifest.permission.BLUETOOTH_CONNECT)
        val requirePermissions = ArrayList<String>()
        for (permission in permissions) {
            if (ActivityCompat.checkSelfPermission(this@MainActivity, permission) != PackageManager.PERMISSION_GRANTED) {
                requirePermissions.add(permission)
            }
        }
        if (!requirePermissions.isEmpty()) {
            ActivityCompat.requestPermissions(this@MainActivity, requirePermissions.toTypedArray(), PERMISSION_REQUEST_CODE)
        } else {
            startService()
        }
    }

    fun startService() {
        startService(Intent(this@MainActivity, AdvertiserService::class.java))
        val config = Config(this)
        config.autoStart = true
        Toast.makeText(this, "Broadcasting started.", Toast.LENGTH_SHORT).show()
    }

    fun stopService() {
        stopService(Intent(this@MainActivity, AdvertiserService::class.java))
        val config = Config(this)
        config.autoStart = false
        Toast.makeText(this, "Broadcasting finished.", Toast.LENGTH_SHORT).show()
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        if (requestCode == PERMISSION_REQUEST_CODE) {
            if (grantResults.size > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                startService()
            } else {
                val onOffSwitch = findViewById<SwitchCompat>(R.id.activate_switch_id)
                onOffSwitch.isChecked = AdvertiserService.isServiceCreated()
            }
        }
    }

    companion object {
        private const val PERMISSION_REQUEST_CODE = 1
    }
}